docs/en/01-how-to-build/snpe.md

```diff
@@ -0,0 +1,187 @@
+# Build for SNPE
+
+It is quit simple to support snpe backend: Client/Servcer mode.
+
+this mode
+
+1. Can split `model convert` and `inference` environments;
+
+- Inference irrelevant matters are done on host
+- We can get the real running results of gpu/npu instead of cpu simulation values
+
+2. Can cover cost-sensitive device, armv7/risc-v/mips chips meet product requirements, but often have limited support for Python;
+
+3. Can simplify mmdeploy installation steps. If you only want to convert snpe model and test, you don't need to compile the .whl package.
+
+## 1. Run inference server
+
+Download the prebuilt snpe inference server package, `adb push` it to the phone and execute.
+Note that **the phone must have a qcom chip**.
+
+```bash
+$ wget https://media.githubusercontent.com/media/tpoisonooo/mmdeploy_snpe_testdata/main/snpe-inference-server-1.59.tar.gz
+...
+$ sudo apt install adb
+$ adb push snpe-inference-server-1.59.tar.gz  /data/local/tmp/
+
+# decompress and execute
+$ adb shell
+venus:/ $ cd /data/local/tmp
+130|venus:/data/local/tmp $ tar xvf snpe-inference-server-1.59.tar.gz
+...
+130|venus:/data/local/tmp $ source export1.59.sh
+130|venus:/data/local/tmp $ ./inference_server
+...
+  Server listening on [::]:50052
+```
+
+At this point the inference service should print all the ipv6 and ipv4 addresses of the device and listen on the port.
+
+tips:
+
+- If `adb devices` cannot find the device, may be:
+  - Some cheap cables can only charge and cannot transmit data
+  - or the "developer mode" of the phone is not turned on
+- If you need to compile the binary by self, please refer to [NDK Cross Compiling snpe Inference Service](../appendix/cross_build_snpe_service.md)
+
+## 2. Build mmdeploy
+
+### 1) Environment
+
+| Matters | Version            | Remarks                |
+| ------- | ------------------ | ---------------------- |
+| host OS | ubuntu18.04 x86_64 | snpe specified version |
+| Python  | **3.6.0**          | snpe specified version |
+
+### 2) Installation
+
+Download [snpe-1.59 from the official website](https://developer.qualcomm.com/qfile/69652/snpe-1.59.0.zip)
+
+```bash
+$ unzip snpe-1.59.0.zip
+$ export SNPE_ROOT=/path/to/snpe-1.59.0.3230
```

Comment:
As long as the first command is unzip, we can use `export SNPE_ROOT=${PWD}/snpe-1.59.0.3230` so user can just copy&run this command.

Reply:
done.